### ISSUE TITLE
SPIR-V: structs as inout parameters don't work

### ISSUE THREAD

**Issue by seppala2** [Microsoft/DirectXShaderCompiler#806]
The optimization pass fails when there is a function that takes a struct as an inout parameter.

For example, the following simplified example fails into an assert in SPIRV-Tools/source/opt/mem_pass.cpp line 103:
```
struct Pixel
{
    float4 color;
};

void applyColor(float3 input, inout Pixel output)
{
    output.color += float4(input, 1.f);
};

float4 PS(in float4 position : SV_Position, in float3 color : COLOR) : SV_Target0
{
    Pixel p;
    p.color = float4(0.0, 0.0, 0.0, 0.0);
    applyColor(color, p);
    return p.color;
}
```

Making the struct an input parameter and returning a new one seems to work.

The arguments given to the compiler were:
-Ges -O3 -WX -Zi -spirv

**Comment by ehsannas** [comment]
Thanks for reporting. I'll start looking into it.

**Comment by antiagainst** [comment]
Fixed by #810
